Add CYS Fiverr Logo Maker CTA

danielwrobert opened the pull request:
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

<!-- You can erase any parts of this template not applicable to your Pull Request. -->

### Changes proposed in this Pull Request:

<!-- If necessary, indicate if this PR is part of a bigger feature. Add a label with the format `focus: name of the feature [team:name of the team]`. -->

<!-- Describe the changes made to this Pull Request and the reason for such changes. -->

Adds the text CTA with a link to the WooCommerce landing page for the Fiverr logo maker in the sidebar navigation on the logo view.

For context, see p7fD6U-eEJ-p2#comment-27868 (internal link).

~Closes # .~

<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

<!-- Include detailed instructions on how these changes can be tested. Review and follow the guide for how to write high-quality testing instructions. -->

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Head over to **WooCommerce > Home > Customize your store**.
2. Click on **Customize your theme** (or **Use the store designer**).
3. Click on **Add your logo**.
4. Confirm the Fiverr CTA text appears below the logo upload input (see screenshot below).
5. Click on the **Fiverr** link in the text and confirm the [Woo-branded Fiverr logo maker landing page](https://www.fiverr.com/logo-maker/woo?afp=&cxd_token=917527_33214203&show_join=true) opens in a new tab.

![CleanShot 2024-06-13 at 22 21 37](https://github.com/woocommerce/woocommerce/assets/481776/e1bb368a-6e22-4341-bf5c-bfd592af3237)

<!-- End testing instructions -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box and supplying data. -->

-   [x] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [ ] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [x] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [x] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

CYS: add CTA to our Fiverr Logo Maker landing page. 

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->

</details>

danielwrobert committed:
Add markup for Fiverr logo maker.

Add text CTA with link to WooCommerce landing page for Fiverr logo
maker in sidebar navigation on the logo view.

danielwrobert committed:
Adjust styles for Fiverr logo maker CTA.

Commit:
Add changefile(s) from automation for the following project(s): woocommerce

Commit:
Add changefile(s) from automation for the following project(s): woocommerce

albarin reviewed (approved):
Looks good!
Just a small comment: the changelog message is empty.

nefeline commented on plugins/woocommerce-admin/client/customize-store/assembler-hub/sidebar/sidebar-navigation-screen-logo.tsx:
```diff
@@ -526,6 +526,24 @@ export const SidebarNavigationScreenLogo = ( {
+							{ __( "DON'T HAVE A LOGO YET?", 'woocommerce' ) }
+						</strong>
+						<p>
+							{ __(
+								'Get a custom logo designed by a professional on ',
+								'woocommerce'
+							) }
+							<a
+								href="https://www.fiverr.com/logo-maker/woo?afp=&cxd_token=917527_33214203&show_join=true"
+								target="_blank"
+								rel="noreferrer"
+							>
+								{ __( 'Fiverr.', 'woocommerce' ) }
```
The text to be translated is broken into two parts here (the sentence "Get a custom logo designed by a professional on Fiverr." is split). This segmentation can lead to translation issues because different languages have different syntactic structures. I would recommend relying on `interpolateComponents` instead. Example:

https://github.com/woocommerce/woocommerce/blob/af2946aa3431d89d0fcbd0748a12628b39b141fc/plugins/woocommerce-admin/client/customize-store/assembler-hub/sidebar/sidebar-navigation-screen-typography.tsx#L181

danielwrobert replied:
Nice catch! I've updated to use `interpolateComponents` in 4675ceaf.

Thanks!

danielwrobert committed:
Update string output to use interpolateComponents

The text in this addition is mixed with string and markup.

This segmentation can lead to translation issues because
different languages have different syntactic structures.

Here, we utilize interpolateComponents to remedy these potential issues.

See
https://github.com/woocommerce/woocommerce/pull/48486#discussion_r1639637477
for context.

danielwrobert commented:
Something odd happened when I rebased with `trunk`. I'm gonna roll back and clean up these unrelated commits.

danielwrobert commented:
> Looks good! Just a small comment: the changelog message is empty.

Thanks @albarin!

Regarding the changelog entry, as I understand it, that is auto-generated when you check off that checkbox under the **Changelog entry** section. But LMK if I'm missing/misunderstanding something!

albarin commented:
> Although I do see that the summary is missing, it's just showing the `Significance` and `Type`! 🤔

Yes, it's auto-generated, but you still need to write the message you want under **Message** on the PR description and it will pick it up and add it in the changelog automatically.

Commit:
Add changefile(s) from automation for the following project(s): woocommerce

danielwrobert commented:
> Yes, it's auto-generated, but you still need to write the message you want under Message on the PR description and it will pick it up and add it in the changelog automatically.

Yeah, I _just_ looked back and noticed I missed that before reading your reply. Thank for catching, I've updated it!
